src/test_typing.py

```diff
@@ -893,6 +893,12 @@ class MyDef(typing.DefaultDict[str, T]): ...
         self.assertIs(MyDef[int]().__class__, MyDef)
         self.assertIs(MyDef[int]().__orig_class__, MyDef[int])
 
+    @skipUnless(sys.version_info >= (3, 3), 'ChainMap was added in 3.3')
+    def test_chainmap_type_erasure_special(self):
+        class MyChain(typing.ChainMap[str, T]): ...
+        self.assertIs(MyChain[int]().__class__, MyChain)
+        self.assertIs(MyChain[int]().__orig_class__, MyChain[int])
```

Comment:
No, that one is intentional. (note that it is already "protected" by ``self.clear_caches()``)